compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -153,6 +153,10 @@ pub trait MirPass<'tcx> {
         to_profiler_name(self.name())
     }
 
+    fn min_mir_opt_level(&self) -> usize {
+        0
+    }
+
     /// Returns `true` if this pass is enabled with the current combination of compiler flags.
     fn is_enabled(&self, _sess: &Session) -> bool {
         true
```

Comment:
This should probably default to `sess.mir_opt_level() >= self.min_mir_opt_level()` now that the method above exists, and then you should be able to replace the `is_enabled` duplication in the passes by using the default.